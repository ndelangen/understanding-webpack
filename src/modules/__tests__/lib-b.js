import * as b from '../lib-b';

describe('sanity', () => {
  it('should exist', () => {
    expect(b).toBeDefined();
  });
  it('should contain the right methods', () => {
    expect(Object.keys(b)).toMatchSnapshot();
  });
});

describe('functional', () => {
  it('should not throw', () => {
    expect(b.bar).not.toThrow();
  });
  it('should work correctly', () => {
    expect(b.bar(1)).toBe(2);
    expect(b.bar(2)).toBe(3);
    expect(b.bar(65)).toBe(66);
  });
  it('should work with invalid data', () => {
    expect(b.bar('1')).toBe('11');
    expect(b.bar('a')).toBe('a1');
    expect(b.bar(NaN)).toBeNaN();
    expect(b.bar(undefined)).toBeNaN();
  });
});
