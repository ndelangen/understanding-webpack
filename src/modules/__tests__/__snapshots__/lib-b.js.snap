// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanity should contain the right methods 1`] = `
Array [
  "foo",
  "bar",
]
`;
